### DISCUSSION TITLE
alarm-settings.yml config webhook，The request message contains the servcie ip and port

### DISCUSSION THREAD

**Discussion by Bert-long** [apache/skywalking#12276] [Q&A] [answered]
rules:
  endpoint_resp_time_rule:
   expression: sum(endpoint_resp_time > 1000) >= 2
   period: 10
   silence-period: 5
   message: Response time of endpoint {name} is more than 1000ms in 2 minutes of last 10 minutes

How can I make the request contain service ip and port information?

**Comment by wu-sheng** [comment]
IP and port are not unavailable. Service is logic concept.

**Reply by Bert-long** [reply]
> IP and port are not unavailable. Service is logic concept.

I'm sorry I didn't describe it well.
I am trying to get which app sent the alert and send the app's IP, port via hook

**Reply by Bert-long** [reply]
> IP and port are not unavailable. Service is logic concept.

After getting the ip, port, I can make the container where the application is located restart or shutdown

**Reply by Bert-long** [reply]
> IP and port are not unavailable. Service is logic concept.

@wu-sheng I checked the official tutorials and github issue and didn't find an answer, I don't know if the alerts can get the app's IP and port, I'm very much looking forward to your reply!

**Reply by wu-sheng** [reply, marked as answer]
I said, there isn't. Service just has a name. Nothing more. Instance has an attribute concept, which may include IP, but no port. Also, both of these of the instance are not available in the alarm engine.

**Reply by Bert-long** [reply]
> I said, there isn't. Service just has a name. Nothing more. Instance has an attribute concept, which may include IP, but no port. Also, both of these of the instance are not available in the alarm engine.

Thank you very much for your answer.

**Comment by Bert-long** [comment]
The problem has been solved.
